### PR TITLE
[13.0][ADD] delivery_sending: New module

### PR DESCRIPTION
**Pull request by victoralmau**
New module

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT40374

### CODE REVIEW

File: delivery_sending/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+This module links the `Sending <https://www.senging.es>`_ booking and tracking
```
Comment (chienandalu) [inline delivery_sending/readme/DESCRIPTION.rst]:
```suggestion
This module links the `Sending <https://www.sending.es>`_ booking and tracking
```

File: delivery_sending/models/sending_request.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+# Copyright 2022 Impulso Diagonal - Javier Colmeiro
+# Copyright 2022 Tecnativa - Víctor Martínez
+import logging
+
+from suds.client import Client
+from suds.sax.text import Raw
```
Comment (chienandalu) [inline delivery_sending/models/sending_request.py:7]: Wasn't `zeep` suitable for the task?

File: delivery_sending/models/sending_request.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+_logger = logging.getLogger(__name__)
+
+
+class SendingRequest(object):
+    """Interface between Sending SOAP API and Odoo recordset
+       Abstract Sending API Operations to connect them with Odoo
+    """
+
+    def __init__(self, carrier):
```
Comment (chienandalu) [inline delivery_sending/models/sending_request.py]: Non blocking, but you know I like to abstract the request class away from any odoo object :grin:

**Review by chienandalu — COMMENTED** [review]
Testing pending. Some little comments for the moment:

File: delivery_sending/models/sending_request.py (hunk 2 above)
Comment (victoralmau) [reply delivery_sending/models/sending_request.py:7]: The current behavior works correctly. If there was a test user for the API I would change it to `zeep`.

File: delivery_sending/models/sending_request.py (hunk 3 above)
Comment (victoralmau) [reply delivery_sending/models/sending_request.py]: Yes, I know, but I think just the opposite.

File: delivery_sending/models/sending_request.py (hunk 2 above)
Comment (chienandalu) [reply delivery_sending/models/sending_request.py:7]: Ok, I didn't notice it was coauthored

File: delivery_sending/tests/test_delivery_sending.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.picking = cls.sale.picking_ids[0]
+        cls.picking.move_lines.quantity_done = 10
+
+    def test_delivery_carrier_sending_integration(self):
+        self.picking.action_confirm()
+        self.picking.action_assign()
+        with self.assertRaises(UserError):
+            self.picking.send_to_shipper()
```
Comment (chienandalu) [inline delivery_sending/tests/test_delivery_sending.py]: IMO this test isn't very useful and adds a unnecesary delay in the CI until the connection timeout is raised.

**Review by chienandalu — APPROVED** [review]
Tested as far as I could. Thanks! :+1:

File: delivery_sending/tests/test_delivery_sending.py (hunk 4 above)
Comment (victoralmau) [reply delivery_sending/tests/test_delivery_sending.py]: Yes, the truth is that without a test user it is not very useful. What do you think @pedrobaeza?

Comment (pedrobaeza) [reply delivery_sending/tests/test_delivery_sending.py]: If there's a connection timeout, then indeed it should go out. The value of this is that the data to send prepare methods at least works. Is there a way to circumvent this?

Comment (victoralmau) [reply delivery_sending/tests/test_delivery_sending.py]: I have changed tests to simulate a complete process.

File: delivery_sending/models/sending_request.py (hunk 3 above)
Comment (pedrobaeza) [reply delivery_sending/models/sending_request.py]: I'm with @chienandalu. This class is not ORM (`models.Model`), but Python one (`object`), so only Python arguments should be passed for eventually extracting this isolatedly from Odoo.

Comment (victoralmau) [reply delivery_sending/models/sending_request.py]: Ok, changes done.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**